### PR TITLE
[storage] explicitly forbid `save_transactions()` from being called concurrently

### PR DESCRIPTION
**Pull request by msmouse**
…rrently

### Description
see discussion on https://github.com/aptos-labs/aptos-core/pull/1502

### Test Plan

existing coverage

### CODE REVIEW

File: storage/aptosdb/src/lib.rs [hunk 1]
```diff
@@ -1295,6 +1297,11 @@ impl DbWriter for AptosDB {
         save_state_snapshots: bool,
     ) -> Result<()> {
         gauged_api("save_transactions", || {
+            let _lock = self
+                .ledger_commit_lock
+                .try_lock()
+                .expect("Concurrent committing detected.");
```
Comment (zekun000) [inline storage/aptosdb/src/lib.rs:1303]: why panic? just wait and lock? this seems more like a debug_assert

Comment (msmouse) [reply storage/aptosdb/src/lib.rs:1303]: We don't support committing an old commit anyway, I'd make a larger noise to be noticed.

Comment (sitalkedia) [reply storage/aptosdb/src/lib.rs:1303]: May be add a comment here to make it clearer?

**Pull request merged**